src/components/common/PieChart/PieChart.jsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect } from 'react';
+import * as d3 from 'd3/dist/d3.min';
+import { CHART_RADIUS, CHART_SIZE } from './constants';
+import { generateArrayOfUniqColors } from './colorsGenerator';
+import './PieChart.css';
+
+export const PieChart = ({ data, dataLegend, pieChartId, dataLegendHeader }) => {
```

Comment:
Hi! One random recommendation here is to try to memoize this component if you can, when react reconciles it may cause this to render again from the parent. Small optimization, but could be good to have in the future if the parents cause re-renders, because basically it will keep re-drawing the pie chart.

Reply:
I think, at the moment this optimization is not helpful. Memoization itself requires memory, so I don't see the point in optimizing prematurely.